src/rdf4cpp/rdf/writer/BufWriter.hpp

```diff
@@ -115,12 +116,12 @@ inline bool write_str(std::string_view str, void *const buffer, Cursor &cursor,
 
         if (max_write == str.size()) [[likely]] {
             break;
-        } else {
-            if (flush(buffer, cursor); cursor.size() == 0) [[unlikely]] {
-                return false;
-            }
+        }
+
+        str.remove_prefix(max_write);
 
-            str.remove_prefix(max_write);
+        if (flush(buffer, cursor, str.size()); cursor.size() == 0) [[unlikely]] {
```

Comment:
remove_prefix moved before flush where `str.size()` is used to assess the remaining capacity requirement. Looks good to me.